tests/integration/test_hyper_transformer.py

```diff
@@ -332,99 +280,87 @@ def test_hypertransformer_field_transformers_multi_column_fields():
     })
     expected_reversed = data.copy()
 
+    # run
     ht = HyperTransformer(field_transformers=field_transformers)
     ht.fit(data)
     transformed = ht.transform(data)
-
-    pd.testing.assert_frame_equal(
-        transformed,
-        expected_transformed
-    )
-
     reverse_transformed = ht.reverse_transform(transformed)
 
+    # assert
+    pd.testing.assert_frame_equal(transformed, expected_transformed)
     pd.testing.assert_frame_equal(expected_reversed, reverse_transformed)
 
 
 def test_single_category():
+    """Test that categorical variables with a single value are supported."""
+    # setup
     ht = HyperTransformer(field_transformers={
         'a': OneHotEncodingTransformer()
     })
     data = pd.DataFrame({
         'a': ['a', 'a', 'a']
     })
 
+    # run
     ht.fit(data)
     transformed = ht.transform(data)
-
     reverse = ht.reverse_transform(transformed)
 
+    # assert
     pd.testing.assert_frame_equal(data, reverse)
 
 
-@pytest.mark.xfail()
 def test_dtype_category():
+    """Test that categorical variables of dtype category are supported."""
+    # setup
     data = pd.DataFrame({'a': ['a', 'b', 'c']}, dtype='category')
 
+    # run
     ht = HyperTransformer()
     ht.fit(data)
+    transformed = ht.transform(data)
+    reverse = ht.reverse_transform(transformed)
 
-    trans = ht.transform(data)
-
-    rever = ht.reverse_transform(trans)
-
-    pd.testing.assert_frame_equal(rever, data)
+    # assert
+    pd.testing.assert_frame_equal(reverse, data)
 
 
-def test_subset_of_columns():
+def test_subset_of_columns_nan_data():
     """HyperTransform should be able to transform a subset of the training columns.
 
     See https://github.com/sdv-dev/RDT/issues/152
     """
-    data = get_input_data_without_nan()
-
+    # setup
+    data = get_input_data()
+    subset = data[[data.columns[0]]].copy()
     ht = HyperTransformer()
     ht.fit(data)
 
-    subset = get_input_data_without_nan()[[data.columns[0]]]
+    # run
     transformed = ht.transform(subset)
     reverse = ht.reverse_transform(transformed)
 
+    # assert
     pd.testing.assert_frame_equal(subset, reverse)
 
 
 def test_with_unfitted_columns():
     """HyperTransform should be able to transform even if there are unseen columns in data."""
-    data = get_input_data_without_nan()
-
+    # setup
```

Comment:
minor: I noticed some files capitalize these. I think we should try and keep consistent, so either have it capitalized everywhere or lower case everywhere